README.md

```diff
@@ -4,22 +4,287 @@
 [![Code Climate](https://codeclimate.com/github/Dhii/module-interface/badges/gpa.svg)](https://codeclimate.com/github/Dhii/module-interface)
 [![Test Coverage](https://codeclimate.com/github/Dhii/module-interface/badges/coverage.svg)](https://codeclimate.com/github/Dhii/module-interface/coverage)
 [![Latest Stable Version](https://poser.pugx.org/dhii/module-interface/version)](https://packagist.org/packages/dhii/module-interface)
-[![This package complies with Dhii standards](https://img.shields.io/badge/Dhii-Compliant-green.svg?style=flat-square)][Dhii]
 
 ## Details
 This package contains interfaces that are useful in describing modules and their attributes and behaviour.
-
-### Interfaces
-- [`ModuleInterface`][] - The interface for a module. A module is an object that represents an
-application fragment. Modules are prepared using `setup()`, which returns a `ServiceProviderInterface` instance that
-the application may consume, and invoked using `run()`.
  
 ### Requirements
 - PHP: 7.1 and up, until 8.
 
     Officially supports at least up to php 7.4.x.
 
+### Interfaces
+- [`ModuleInterface`][] - The interface for a module. A module is an object that represents an
+application fragment. Modules are prepared using `setup()`, which returns a `ServiceProviderInterface` instance that
+the application may consume, and invoked using `run()`, consuming the application's DI container.
+- [`ModuleAwareInterface`][] - Something that can have a module retrieved.
+- [`ModuleExceptionInterface`][] - An exception thrown by a module.
+
+### Usage
+#### Module Package
+In your module's pacakge, create a file that returns a module factory. This factory MUST return an instance
+of `ModuleInterface` from this pacakge. By convention, this file has
+the name `module.php`, and is located in the root directory. Below is a very basic example. In real life,
```

Comment:
Because you cannot and should not standardize the module's constructor, as every module can have their own requirements. But you can and should standardize a factory.